synapse/models/dns.py

```diff
@@ -1,75 +1,111 @@
+                    ('ipv6', ('inet:ipv6', {}), {'ro': 1,
+                         'doc': 'The IPv4 address queried for its DNS PTR record.'}),
+                    ('fqdn', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain returned in the PTR record.'}),
+                    # FIXME - add Seen:min / seen:max
+                )),
+                ('inet:dns:ns', {}, (
+                    ('zone', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain queried for its DNS NS record.'}),
+                    ('ns', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain returned in the NS record.'}),
+                    # FIXME - add Seen:min / seen:max
+                )),
+                ('inet:dns:cname', {}, (
+                    ('fqdn', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain queried for its CNAME record.'}),
+                    ('cname', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain returned in the CNAME record.'}),
+                    # FIXME - add Seen:min / seen:max
+                )),
+                ('inet:dns:mx', {}, (
+                    ('fqdn', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain queried for its MX record.'}),
+                    ('mx', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain returned in the MX record.'}),
+                    # FIXME - add Seen:min / seen:max
+                )),
+                ('inet:dns:soa', {}, (
+                    ('fqdn', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain queried for its SOA record.'}),
+                    ('ns', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain (MNAME) returned in the SOA record.'}),
+                    ('email', ('inet:email', {}), {'ro': 1,
+                         'doc': 'The email address (RNAME) returned in the SOA record.'}),
+                    # FIXME - add Seen:min / seen:max
+                )),
+                ('inet:dns:txt', {}, (
+                    ('fqdn', ('inet:fqdn', {}), {'ro': 1,
+                         'doc': 'The domain queried for its TXT record.'}),
+                    ('txt', ('str', {}), {'ro': 1,
+                         'doc': 'The string returned in the TXT record.'}),
+                    # FIXME - add Seen:min / seen:max
+                )),
+            )
+        }
+        omodl = {
             'types': (
                 ('inet:dns:look', {
                     'subof': 'guid',
                     'doc': 'The instance (point-in-time) result of a DNS record lookup.'}),
 
-                ('inet:dns:a', {
-                    'subof': 'sepr',
-                    'sep': '/',
-                    'fields': 'fqdn,inet:fqdn|ipv4,inet:ipv4',
-                    'doc': 'The result of a DNS A record lookup.',
-                    'ex': 'vertex.link/1.2.3.4'}),
```

Comment:
I think we should keep the examples

Reply:
all the examples were for sepr's. how do we wanna handle examples for comps?

Reply:
That's a great question. Do we want the `ex` field to be a normable valu? or a repr?

Reply:
per discussion - storm runtime parsable

Reply:
added examples to all comp types